src/pymor/algorithms/dmd.py

```diff
@@ -0,0 +1,270 @@
+
+from pymor.core.defaults import defaults
+from pymor.vectorarrays.interface import VectorArray
+from pymor.vectorarrays.numpy import NumpyVectorSpace
+from pymor.algorithms.svd_va import method_of_snapshots, qr_svd
+from pymor.algorithms.gram_schmidt import gram_schmidt
+
+from pymor.operators.constructions import VectorArrayOperator
+from pymor.algorithms.genericsolvers import lsqr
+
+
+@defaults('svd_method')
+def dmd(A=None, XL=None, XR=None, target_rank=None, dt=1, modes='exact', order=True, svd_method='qr_svd'):
+    """Dynamic Mode Decomposition.
+
+    See Algorithm 1 and Algorithm 2 in :cite:`TRLBK14`.
+
+    Parameters
+    ----------
+    A  :  optional
+        The |VectorArray| for which the DMD Modes are to be computed.
+        If A is used, XL and XR should not be used.
+
+    XL  :  optional
+        The |VectorArray| of the left Snapshot series A[:-1].
+    XR  :  optional
+        The |VectorArray| of the right Snapshot series A[1:].
+        If XL and XR are given, A should not be given.
+
+    target_rank : int
+        Number of DMD Modes to be computed. If None target_rank = len(A).
+
+    dt : scalar, optional (default: 1)
+        Factor specifying the time difference between the observations.
+        Used if the input data is a timeseries in continuous time.
+
+    modes : str `{'standard', 'exact', 'exact_scaled'}`
+        - 'standard' : uses the standard definition to compute the dynamic modes, `Wk = U * EVECS`.
+        - 'exact' : computes the exact dynamic modes, `Wk = XR * V * Sigma_inv * EVECS`.
+        - 'exact_scaled' : computes the exact dynamic modes,
+                           `Wk = (1/EVALS) * XR * V * Sigma_inv * EVECS`.
+
+    order : bool `{True, False}`
+        True: return modes sorted.
+
+    svd_method
+        Which SVD method from :mod:`~pymor.algorithms.svd_va` to use
+        (`'method_of_snapshots'` or `'qr_svd'`).
+
+
+    Returns
+    -------
+    Wk : VectorArray
+        |VectorArray| containing the dynamic modes.
+
+    omega : array_like
+        Time scaled eigenvalues: `ln(l)/dt`.
+    """
+
+    assert modes in ('exact', 'standard', 'exact_scaled')
+    assert A is None or (XL is None and XR is None)
+    assert not(A is None and (XL and XR is None))
+    assert isinstance(A, VectorArray) or A is None
+    assert isinstance(XL, VectorArray) or XL is None
+    assert isinstance(XR, VectorArray) or XR is None
+    assert A is None or (target_rank is None or target_rank <= len(A))
+    assert XL is None or (target_rank is None or target_rank < len(XL))
+    assert XL is None and XR is None or len(XL) == len(XR)
+    assert order in (True, False)
+    assert svd_method in ('qr_svd', 'method_of_snapshots')
+
+    # XL = x0, ..., xm-1  ; XR = x1, ..., xm
+    if A is not None:
+        XL = A[:-1]
+        XR = A[1:]
+
+    rank = len(XL) if target_rank is None else target_rank
+    svd = qr_svd if svd_method == 'qr_svd' else method_of_snapshots
+
+    print('SVD of XL...')
```

Comment:
Can you replace all print statements in the library code by corresponding logging code like, for instance, in `pymor.algorithms.gram_schmidt.gram_schmidt`?